### ISSUE TITLE
if graph final have many leaf tasks, can transwatp get the final result?

### ISSUE THREAD

**Issue by rrjia** [bloomen/transwarp#31]
because the program finally use the last task excute

std::shared_ptr<tw::task<long>> build_graph() {
    auto task0 = tw::make_task(tw::root, func0);
    auto task1 = tw::make_task(tw::root, func1);
    auto task2 = tw::make_task(tw::consume, func2, task0, task1);
    auto task3 = tw::make_task(tw::root, func3);
    auto task4 = tw::make_task(tw::consume, func4, task2, task3);    
    return task4;
}

void calculate_via_transwarp(tw::task<long>& task) {
    tw::parallel executor{4}; 
    task.schedule_all(executor);
    long result = task.get();  
    std::cout << "transwarp result is : " << result << std::endl; 
}

**Comment by bloomen** [comment]
Not sure I understand your question. Can you rephrase?

**Comment by rrjia** [comment]
if task graph is as follow:
          ------B
A-----|
          ------C

**Comment by rrjia** [comment]
<img width="245" alt="0" src="https://user-images.githubusercontent.com/13146304/104552197-4c85d880-5673-11eb-8da7-6748768b7350.png">

**Comment by bloomen** [comment]
I assume you want to get the results from both B and C. In that case you need to create a single final task D that takes B and C as input. Then just forward B and C results within D. The graph is scheduled via D.
